lib/deps/upsert.js

```diff
@@ -1,7 +1,55 @@
 'use strict';
 
-var upsert = require('pouchdb-upsert').upsert;
+var Promise = require('lie');
```

Comment:
We should pull from deps/promise so that we use the native `Promise` if it's available.